### PR TITLE
test(consolidation,inspect): orchestrator + serde roundtrip tests; fix DumpFormat doc

### LINKED ISSUE #127: [super-qa] medium: DumpFormat::Sqlite doc contradicts implementation
## Finding: Stale Doc on DumpFormat::Sqlite

- **ID:** M20
- **Severity:** medium
- **Category:** documentation
- **Location:** src/inspect/types.rs:143-144

## Description

Doc says "file-backed engines only" but VACUUM INTO works for in-memory engines too (SQLite 3.27+). Engine.rs docs say it works for both.

## Suggested Approach

Update the types.rs doc to match engine.rs.

### LINKED ISSUE #129: [super-qa] medium: missing tests for inspect types, consolidation orchestrator, bootstrap orchestrator
## Finding: Missing Tests for Orchestration Code

- **IDs:** M-inspect-types, M-consolidation-mod, M-bootstrap-mod
- **Severity:** medium
- **Category:** testing

## Description

Several orchestration modules lack inline tests:

1. **src/inspect/types.rs** — 15+ structs with Serialize/Deserialize but no serde roundtrip tests. EngineSnapshot is the import/export format.
2. **src/consolidation/mod.rs** — consolidate() orchestrator reads config, runs 3 passes in transaction. No unit tests for config read/write or partial failure.
3. **src/bootstrap/mod.rs** — bootstrap_session_inner() has savepoint management, rollback on failure. No inline tests for edge cases.

## Suggested Approach

Add targeted tests for each: serde roundtrips for types, savepoint rollback for bootstrap, config persistence for consolidation.

### PR DESCRIPTION
**Pull request by dutiona**
## Summary

Adds test coverage to two previously-untested modules (#129, re-scoped to consolidation + inspect-types) and fixes a stale doc comment (#127).

### #129 — new tests

**`src/consolidation/mod.rs`** — 7 unit tests for the `consolidate()` orchestrator (was 0):
- **Config read/write** of `last_consolidated_at`: written after a successful run, read to scope dedup to new facts only, and a malformed value surfaces as `MemoryError::Migration`.
- **3-pass transaction**: dedup → cluster → global all run in one transaction; verified stats and the cluster/global summary outputs.
- **Partial-failure ROLLBACK**: a `FailingGenerator` in the cluster pass aborts the whole consolidation — dedup expirations are undone, no summaries persist, and the watermark is **not** advanced.
- Plus an empty-engine no-op case.

**`src/inspect/types.rs`** — 18 serde round-trip tests (was 0) covering every `Serialize`/`Deserialize` type in the import/export wire format. Each asserts `serialize → deserialize == original`; the two non-`PartialEq` snapshot types (`EngineSnapshot`, `EngineStatistics`-snapshot path) assert JSON stability instead. Also covers the `#[serde(default)]` back-compat path for pre-v8 snapshots missing `lineage`.

### #127 — doc fix

`DumpFormat::Sqlite` doc claimed `VACUUM INTO` was "file-backed engines only". It works for in-memory engines too (SQLite 3.27+, per #80, and `engine.rs` / `docs/usage/inspection.md` already say both). Reworded to "file-backed and in-memory engines".

## Gates

- `cargo build --workspace` ✅
- `cargo test -p memory-engine -p memory-engine-mcp -p memory-engine-embed` ✅ (core: 691 passed, incl. 25 new tests)
- `cargo clippy --workspace --all-targets` — DELTA-CLEAN: zero new warnings from the changed files (pre-existing pedantic/nursery warnings tracked in #539)
- `cargo fmt --check` ✅

## Pre-existing breakage (not introduced here)

`cargo test --workspace` fails to compile `memory-engine-cli`'s `cli_integration` test (`MemoryEngine::open` was removed by the typestate-builder refactor #556). Untouched by this PR — surfaced for visibility.

Fixes #129
Fixes #127

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/consolidation/mod.rs [hunk 1]
```diff
@@ -112,3 +112,290 @@ pub fn consolidate(
         expired_ids,
     ))
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use chrono::Duration;
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/consolidation/mod.rs]: Addressed in e98aa89. Resolving.

File: src/consolidation/mod.rs [hunk 2]
```diff
@@ -112,3 +112,290 @@ pub fn consolidate(
...
+        }
+    }
+
+    /// Insert an active fact, returning its id.
+    fn insert_fact(conn: &Connection, content: &str, embedding: Vec<f32>, importance: f64) -> i64 {
+        let store = FactStore::new(conn, DIM);
+        store
+            .insert(&NewFact {
+                content: content.into(),
+                content_hash: blake3::hash(content.as_bytes()).to_hex().as_str()[..32].to_string(),
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/consolidation/mod.rs]: Addressed in e98aa89. Resolving.

File: src/consolidation/mod.rs [hunk 3]
```diff
@@ -112,3 +112,290 @@ pub fn consolidate(
...
+        );
+    }
+
+    #[test]
+    fn watermark_read_scopes_dedup_to_new_facts() {
+        let conn = open_memory().unwrap();
+        init_schema(&conn).unwrap();
+
+        // Seed a watermark in the future so EVERY existing fact predates it.
+        let future = Utc::now() + Duration::days(1);
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/consolidation/mod.rs]: Addressed in e98aa89. Resolving.

File: src/inspect/types.rs [hunk 4]
```diff
@@ -228,3 +228,447 @@ pub struct StorageStats {
...
+            embed_dim: 4,
+            facts: vec![sample_fact()],
+            edges: vec![sample_edge()],
+            summaries: vec![sample_summary()],
+            scopes: vec![sample_scope()],
+            events: vec![sample_event()],
+            lineage: vec![sample_lineage()],
+            config,
+        };
+        // EngineSnapshot is not PartialEq; assert JSON stability instead.
```
[the remark this answers is not in the document]
Comment (dutiona) [reply src/inspect/types.rs]: Addressed in e98aa89. Resolving.

[the remark this answers is not in the document]
Comment (dutiona) [reply src/inspect/types.rs]: Correction to my prior auto-reply: this suggestion was **declined**, not applied — and here's why. Deriving `PartialEq` + a structural roundtrip (`assert_eq!(deserialize(serialize(x)), x)`) **fails**: `PromotionProvenance.lineage_id` is a denormalized field re-derived from its parent `LineageSnapshotEntry` on load, so serialize→deserialize is not value-identity (verified empirically — the sample's `lineage_id: 0` becomes `21` after a roundtrip). The existing JSON-stability assertion (re-serialization is byte-stable) is the correct check for this format; kept it and documented the reason in e98aa89. (Aside: the denormalized `lineage_id` duplicate in the export type is a minor design smell, noted for the maintainer.)

**Comment by dutiona** [comment]
## Adversarial review summary (replaces /super-review)

Reviewed by **3 test-lens reviewers** — **3/3 approve, 0 blockers**:

| Lens         | Verdict    | Note                                                                                                                                                                  |
| ------------ | ---------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| coverage     | ✅ APPROVE | consolidation: 7 orchestrator tests (watermark r/w, 3-pass transaction, **partial-failure rollback**); inspect: 18 serde roundtrip tests + pre-v8 lineage back-compat |
| test-quality | ✅ APPROVE | assertions discriminating, rollback test verifies state actually rolled back                                                                                          |
| correctness  | ✅ APPROVE | #127 doc fix matches `engine.rs` reality (VACUUM INTO works in-memory)                                                                                                |

**Gemini bot (4 threads, all resolved in `e98aa89`):** `Duration`→`TimeDelta` (×2, deprecated API), redundant manual BLAKE3 `content_hash`→`String::new()` (FactStore recomputes it). The `derive(PartialEq)`+structural-roundtrip suggestion was **declined with rationale** — `PromotionProvenance.lineage_id` is denormalized (re-derived on load), so serialize→deserialize is not value-identity; JSON-stability is the correct assertion (documented).

> Gate note: `cargo test/clippy --workspace` is blocked by the **pre-existing** `cli_integration` break (#564, the #556 fallout) — **zero** new lints/failures from this PR; core 778 tests + core clippy clean. Resolves once #568 (which fixes #564) merges.
